nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/main/java/org/apache/nifi/elasticsearch/ElasticSearchClientServiceImpl.java

```diff
@@ -0,0 +1,255 @@
+            URL u = new URL(hostsSplit[x]);
+            hh[x] = new HttpHost(u.getHost(), u.getPort(), u.getProtocol());
+        }
+
+        RestClientBuilder builder = RestClient.builder(hh)
+                .setHttpClientConfigCallback(httpClientBuilder -> {
+                    if (sslService != null) {
+                        try {
+                            KeyStore keyStore = KeyStore.getInstance(sslService.getKeyStoreType());
+                            KeyStore trustStore = KeyStore.getInstance("JKS");
+
+                            try (final InputStream is = new FileInputStream(sslService.getKeyStoreFile())) {
+                                keyStore.load(is, sslService.getKeyStorePassword().toCharArray());
+                            }
+
+                            try (final InputStream is = new FileInputStream(sslService.getTrustStoreFile())) {
+                                trustStore.load(is, sslService.getTrustStorePassword().toCharArray());
+                            }
+
+                            final KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory
+                                    .getDefaultAlgorithm());
+                            kmf.init(keyStore, sslService.getKeyStorePassword().toCharArray());
+                            final TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory
+                                    .getDefaultAlgorithm());
+                            tmf.init(keyStore);
+                            SSLContext context1 = SSLContext.getInstance(sslService.getSslAlgorithm());
+                            context1.init(kmf.getKeyManagers(), tmf.getTrustManagers(), new SecureRandom());
+                            httpClientBuilder = httpClientBuilder.setSSLContext(context1);
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+
+                    if (username != null && password != null) {
+                        final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+                        credentialsProvider.setCredentials(AuthScope.ANY,
+                                new UsernamePasswordCredentials(username, password));
+                        httpClientBuilder = httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+                    }
+
+                    return httpClientBuilder;
+                })
+                .setRequestConfigCallback(requestConfigBuilder -> {
+                    try {
+                        requestConfigBuilder.setConnectTimeout(connectTimeout);
+                        requestConfigBuilder.setSocketTimeout(readTimeout);
+                        return requestConfigBuilder;
+                    } catch (Exception ex) {
+                        try {
+                            PrintWriter writer = new PrintWriter(new FileWriter("/tmp/out.log"));
```

Comment:
Why write to "/tmp/out.log"?

Reply:
Something I wrote to make it easier to dump the raw query. Shouldn't be in there.